mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -979,17 +954,36 @@ void CLeaderboardsTimes::ApplySchemeSettings(IScheme* pScheme)
     }
 }
 
-void CLeaderboardsTimes::OnConfirmDeleteReplay(int itemID, const char* file)
+void CLeaderboardsTimes::OnConfirmDeleteReplay(const char* file)
 {
     if (file)
     {
         g_pFullFileSystem->RemoveFile(file, "MOD");
-        m_bTimesNeedUpdate[TIMES_LOCAL] = true;
-        if (m_pLocalLeaderboards->RemoveItem(itemID))
+
+        bool bFound = false;
+
+        char filenameInVec[MAX_PATH];
+        FOR_EACH_VEC(m_vLocalTimes, i)
+        {
+
+            Q_snprintf(filenameInVec, MAX_PATH, "%s/%s-%s%s", RECORDING_PATH, m_vLocalTimes[i]->GetMapName(),
+                       m_vLocalTimes[i]->GetRunHash(), EXT_RECORDING_FILE);
+
+            if (FStrEq(filenameInVec, file))
+            {
+                m_vLocalTimes.Remove(i);
+                bFound = true;
+                break;
+            }
+        }
+        if (!bFound)
         {
-            g_pMOMRunCompare->LoadComparisons();
-            FillLeaderboards(false);
+            Warning("Error while deleting replay: File not found! This probably means that the replay file name was invalid.");
+            return;
         }
+        // MOM_TODO this is still synchoronous so it will lag
+        g_pMOMRunCompare->LoadComparisons();
+        FillLeaderboards(false);
```

Comment:
I can do you one better:

`ConvertLocalTimes` should store the replay pointer inside of the item data via something like:
`kvLocalTimeFormatted->SetPtr("replayPtr", t);`

You can then feed this into the ConfirmDeleteReplay context menu item in `OnItemContextMenu` since it fetches this exact KV inside the variable `selectedRun`. You should still pass `runName` and `itemID` as those will handle removing it from the list of runs on the replay and deleting the file, but you should also add a nice forward:
`pMessage->SetPtr("replayPtr", selectedRun->GetPtr("replayPtr"));`

You can then change the type of the `ContextDeleteReplay` message to be:
`MESSAGE_FUNC_PARAMS(OnContextDeleteReplay, "ContextDeleteReplay", pData);`

And then, finally, in this function, do:
`m_vLocalTimes.FindAndRemove(pData->GetPtr("replayPtr"));`

You'll also need to convert the function to take the `KeyValues *pData` from the macro, and then get all of the data manually (e.g. `const char *file = pData->GetString("runName");` ) but it will be much cleaner in the end.

Reply:
The reason to do this is to tie in with my previous comment about the `RemoveAll`, because I don't think that's a good idea unless the map is being shut down.

Reply:
I'll implement this however one small thing, `runName` should really go because it gets constructed via the mapname-hast.mrf logic in ConvertLocalTimes

Reply:
Also I've run into another issue, mom_replay_play expects the path without the replays/ part. 
I've settled on setting runName to `replayPtr->GetFilePath()[Q_strlen(RECORDING_PATH)]` since we are on the local panel and there is no way for a file not to be in that folder, unless you have a better idea.

Reply:
Yeah that's fine.